### ISSUE TITLE
zsh completion is broken

### ISSUE THREAD

**Issue by monester** [linkerd/linkerd2#3753]
## Bug Report
zsh completion is  failing with error:
```
_arguments:comparguments:325: invalid option definition: (-l --linkerd-namespace)-l[Namespace in which Linkerd is installed [$LINKERD_NAMESPACE]]:
```
It is happening because `[$LINKERD_NAMESPACE]` is not escaped.

### What is the issue?
Unable to use autocompletion for linkerd in zsh

### How can it be reproduced?
Install zsh, add completion by running `source <(linkerd completion zsh)`, try autocomple any linkerd command

### Logs, error output, etc
None

#### `linkerd check` output
Not related

### Environment
- Linkerd version: Client version: edge-19.11.3
- Zsh version: zsh 5.7.1 (x86_64-redhat-linux-gnu)

### Possible solution
There is PR opened in cobra package to fix issue - https://github.com/spf13/cobra/pull/899/files but not yet merged.

**Comment by grampelberg** [comment]
Looks like we're still waiting on the upstream PR to merge.

**Comment by olix0r** [comment]
Given the state of https://github.com/spf13/cobra/pull/899, should we consider cobra abandoned? 

Can we pull in a fix without cobra? Or should we plan to move off of it?

**Comment by vkudrik** [comment]
Any updates on it?

**Comment by grampelberg** [comment]
@vkudrik see upstream issue.

**Comment by alpeb** [comment]
Update: we have updated to cobra 1.0.0 recently (to add fish completion), but that still doesn't have the zsh completion fix. They did fix that in their master branch afterwards but they haven't cut a release including it (last release was in April :man_shrugging: )

**Comment by alpeb** [comment]
Duplicate of #3999
